BraintreeCore/src/main/java/com/braintreepayments/api/AnalyticsClient.kt

```diff
@@ -145,21 +145,23 @@ internal class AnalyticsClient @VisibleForTesting constructor(
         timestamp: Long,
         authorization: Authorization?
     ) {
-        if (lastKnownAnalyticsUrl == null || authorization == null) {
+        if (authorization == null) {
             return
         }
         val metadata = deviceInspector.getDeviceMetadata(context, sessionId, integration)
         val event = AnalyticsEvent("android.crash", timestamp)
         val events = listOf(event)
         try {
             val analyticsRequest = serializeEvents(authorization, events, metadata)
-            httpClient.post(
-                lastKnownAnalyticsUrl,
-                analyticsRequest.toString(),
-                null,
-                authorization,
-                HttpNoResponse()
-            )
+            lastKnownAnalyticsUrl?.let { analyticsUrl ->
```

Comment:
i would move the `lastKnownAnalyticsUrl?.let` right after you check for `authorization == null`, because you are doing a bunch of operations that dont matter if `lastKnownAnalytics is null`